### DISCUSSION TITLE
Get Reference Using DOI and / or ISBN

### DISCUSSION THREAD

**Discussion by blueray453** [papis/papis#871] [Q&A] [answered]
What i am trying to do is, search with book title and get a json like:

```
{
    "author_list": [
      {
        "given": "Nassim Nicholas",
        "family": "Taleb",
        "affiliation": []
      }
    ],
        "title": "Study on AI",
        "year": 2023,
        "edition": 2,
        "volume":,
       "issue":,
        "publisher": "Journal of AI Research",
        "doi": "10.1234/jair.2023.001",
       "isbn": "2321321321312",
       "issn":,
    }
```

The best result i got is from `papis explore isbn -q "Antifragile: Things That Gain from Disorder" export --format json | jq`.

```
> papis explore isbn -q "Antifragile: Things That Gain from Disorder" export --format json | jq
[INFO] isbn: Looking up ISBN documents...
[INFO] isbn: Found 1 documents.
[
  {
    "author_list": [
      {
        "family": "Taleb",
        "given": "Nassim Nicholas"
      }
    ],
    "isbn": "9780812979688",
    "isbn-13": "9780812979688",
    "title": "Antifragile - Things That Gain From Disorder",
    "publisher": "Random House Trade Paperbacks",
    "year": "2014",
    "language": "en",
    "author": "Taleb, Nassim Nicholas"
  }
]

papis explore isbn -q "9780812979688" export --format json | jq
[INFO] isbn: Looking up ISBN documents...
[INFO] isbn: Found 1 documents.
[
  {
    "author_list": [
      {
        "family": "Taleb",
        "given": "Nassim Nicholas"
      }
    ],
    "isbn": "9780812979688",
    "isbn-13": "9780812979688",
    "title": "Antifragile - Things That Gain From Disorder",
    "publisher": "Random House Trade Paperbacks",
    "year": "2014",
    "language": "en",
    "author": "Taleb, Nassim Nicholas"
  }
]
```
However, I noticed that **it does not give the edition** information. The following is actually for the 2nd edition of Artificial Intelligence - A Modern Approach.

```
> papis explore isbn -q "9780137903955" export --format json | jq
[INFO] isbn: Looking up ISBN documents...
[INFO] isbn: Found 1 documents.
[
  {
    "author_list": [
      {
        "family": "Russell",
        "given": "Stuart Jonathan"
      },
      {
        "family": "Norvig",
        "given": "Peter"
      }
    ],
    "isbn": "9780137903955",
    "isbn-13": "9780137903955",
    "title": "Artificial Intelligence - A Modern Approach",
    "publisher": "",
    "year": "2003",
    "language": "en",
    "author": "Russell, Stuart Jonathan and Norvig, Peter"
  }
]
```
What can we do to the the edition?

Is there anything similar for doi where i use the doi to search for the reference.

**Comment by alexfikl** [comment]
We're currently using [isbnlib](https://github.com/xlcnd/isbnlib) to query for information using the ISBN. Unfortunately, it seems like it doesn't return any information regarding the editions..

`isbnlib` can query different services: `goob` for Google books, `openl` for [OpenLibrary.org](https://openlibrary.org/) and `wiki` for Wikipedia books (see `papis explore isbn --help`). You can choose which one to use with
```
papis explore --service goob --query <MY_QUERY>
```
but at least for the Norvig book, none of the services seem to give any information about the edition :\ Do you happen to know of any source that also returns the editions?

> Is there anything similar for doi where i use the doi to search for the reference.

I know some science books (especially on Springer) also have a DOI, but in general the ISBN is used as an identifier for books.

**Reply by blueray453** [reply]
> Do you happen to know of any source that also returns the editions?

Please let me search for it.

**Reply by blueray453** [reply]
In https://www.zotero.org/ahmad.ismail/library there is a option, "Add by Identifier". I gave the ISBN and it returned data including edition.

**Reply by alexfikl** [reply, marked as answer]
Hm, thank you! Will have to look a bit at what magic Zotero does to get better information about the editions.

We can definitely add an `Explore by identifier` explorer to papis as well to make it easier to use for some obvious cases. If you have some python skills, contributions are always welcome :grin:

**Reply by blueray453** [reply]
Thank you very much for considering. In `Explore by identifier` we can use URL, ISBNs, DOIs, PMIDs, or arXiv IDs. It identifies it probably by using regex and return the result.
